include/inexor/vulkan-renderer/wrapper/swapchain.hpp

```diff
@@ -0,0 +1,86 @@
+#pragma once
+
+#include "inexor/vulkan-renderer/availability_checks.hpp"
+#include "inexor/vulkan-renderer/settings_decision_maker.hpp"
+
+#include <spdlog/spdlog.h>
+#include <vulkan/vulkan.h>
+
+#include <cassert>
+#include <vector>
+
+namespace inexor::vulkan_renderer::wrapper {
+
+class Swapchain {
+private:
+    // TODO: Move members which don't need to be members!
+    VkDevice device;
+    VkPhysicalDevice graphics_card;
+    VkSurfaceKHR surface;
+    VkSwapchainKHR swapchain;
+    VkSwapchainKHR old_swapchain;
+    VkSurfaceFormatKHR surface_format;
+    VkExtent2D extent;
+
+    std::vector<VkImage> swapchain_images;
+    std::vector<VkImageView> swapchain_image_views;
+    std::uint32_t images_in_swapchain_count;
+    bool vsync_enabled;
+
+    /// @brief
+    /// @param
+    /// @param
+    void setup_swapchain(std::uint32_t &window_width, std::uint32_t &window_height);
+
+public:
+    /// Delete the copy constructor so swapchains are move-only objects.
+    Swapchain(const Swapchain &) = delete;
+    Swapchain(Swapchain &&other) noexcept;
+
+    /// Delete the copy assignment operator so swapchains are move-only objects.
+    Swapchain &operator=(const Swapchain &) = delete;
+    Swapchain &operator=(Swapchain &&) noexcept = delete;
+
+    /// @brief
+    /// @note We must pass width and height as call by reference!
+    Swapchain(const VkDevice device, const VkPhysicalDevice graphics_card, const VkSurfaceKHR surface, std::uint32_t &window_width,
+              std::uint32_t &window_height, const bool enable_vsync);
+
+    ~Swapchain();
+
+    /// @brief The swapchain needs to be recreated if it has been invalidated.
+    /// @note We must pass width and height as call by reference!
+    /// This happens for example when the window gets resized.
+    void recreate(std::uint32_t &window_width, std::uint32_t &window_height);
+
+    [[nodiscard]] const auto get_swapchain_ptr() const {
```

Comment:
We should probably make this explicit, I think `auto` can be replaced with `VkSwapchainKHR *`

Reply:
Done.